ui/components/component-library/text-field/text-field.js

```diff
@@ -0,0 +1,85 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+import { SIZES } from '../../../helpers/constants/design-system';
+
+import { Icon, ICON_NAMES } from '../icon';
+
+import { TextFieldBase } from '../text-field-base';
+
+export const TextField = ({
+  className,
+  showClear,
+  clearIconProps,
+  clearButtonProps,
+  rightAccessory,
+  value: valueProp,
+  onChange,
+  onClear,
+  inputProps,
+  ...props
+}) => {
+  const [value, setValue] = useState(valueProp || '');
+  const handleOnChange = (e) => {
+    setValue(e.target.value);
+    onChange && onChange(e);
+  };
+  const handleClear = (e) => {
+    setValue('');
+    clearButtonProps?.onClick && clearButtonProps.onClick(e);
+    onClear && onClear(e);
+  };
+  return (
+    <TextFieldBase
+      className={classnames('mm-text-field', className)}
+      value={value}
+      onChange={handleOnChange}
+      rightAccessory={
+        value && showClear ? (
+          <>
+            {/* replace with ButtonIcon */}
+            <button
+              className="mm-text-field__button-clear"
+              {...clearButtonProps}
+              onClick={handleClear}
+            >
+              <Icon
+                className="mm-text-field__button-clear__icon"
```

Comment:
We shouldn't do nested block names like this, that doesn't match BEM syntax. This should be buton-clear-icon, and should be treated as a separate block

Reply:
Right this makes sense. The `Icon` doesn't care about its context within the `TextField` but rather its context inside of the button. So it should be named accordingly to the button rather than the input. Great catch thanks @brad-decker I need to read up on BEM again 🙇 😓 Eventually this will be replaced with a `ButtonIcon` component that handles the icon styling. I also just realized that there is no need for the icon to have a className because no styles are being attached to it anyway. 🤦

Reply:
Updated here along with @darkwing's suggestions: https://github.com/MetaMask/metamask-extension/pull/16105/commits/f09595e0bd7d031531ce9d4395a6a14bdac522c6
Fixing up my sloppy code and adding test here: https://github.com/MetaMask/metamask-extension/pull/16105/commits/cb5e62dedc2142a6658feaa2f4a57096aa0d8225